server/routes/README.md

```diff
@@ -0,0 +1,318 @@
+# API Routes
+An exhaustive guide to all the API routes and what to send/expect from them.  
+Path parameters are denoted by curly braces `{ }`.  
+Query parameters are denoted by angle brackets `< >`.  
+Body parameters should be passed in the body of the request, as form data.
+
+**A user must be logged in to use any of these routes**, except for user create.  
```

Comment:
We need to specify the format of the JSON to send along with the http error code (403) if the user is not logged in.
For now I'm doing something like this:
```
{
  "failure": "you are not logged in"
}
```
Is that ok?

Reply:
use 401 for not logged in. 403 is if you're logged in but not allowed to see the resource (e.g. a logged in user trying to access a group they're not a part of). see this [stackoverflow](https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses)

as far as json, failure --> err? thoughts?